test/move_effect_toxic_spikes.c

```diff
@@ -0,0 +1,120 @@
+#include "global.h"
+#include "test_battle.h"
+
+SINGLE_BATTLE_TEST("Toxic Spikes inflicts poison")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_TOXIC_SPIKES); }
+        TURN { SWITCH(opponent, 1); }
+    } THEN {
+        EXPECT(gSideTimers[B_SIDE_OPPONENT].toxicSpikesAmount == 1);
+        EXPECT(opponent[0].status1 & STATUS1_POISON);
+    }
+}
+
+SINGLE_BATTLE_TEST("Toxic Spikes inflicts toxic poison")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_TOXIC_SPIKES); }
+        TURN { MOVE(player, MOVE_TOXIC_SPIKES); }
+        TURN { SWITCH(opponent, 1); }
+    } THEN {
+        EXPECT(gSideTimers[B_SIDE_OPPONENT].toxicSpikesAmount == 2);
+        EXPECT(opponent[0].status1 & STATUS1_TOXIC_POISON);
+    }
+}
```

Comment:
Following a similar pattern to my suggestion for spikes, you could consider `PARAMETRIZE`ing these:
```c
SINGLE_BATTLE_TEST("Toxic Spikes inflicts poison")
{
    u32 amountUsed;
    u32 status1;
    GIVEN {
        PLAYER(SPECIES_WOBBUFFET);
        OPPONENT(SPECIES_WOBBUFFET);
        OPPONENT(SPECIES_WOBBUFFET);
    } WHEN {
        u32 cnt;
        for (cnt = 0; cnt < amountUsed; ++cnt)
            TURN { MOVE(player, MOVE_TOXIC_SPIKES); }
        TURN { SWITCH(opponent, 1); }
    } THEN {
        EXPECT(opponent[0].status1 & status1);
        EXPECT(gSideTimers[B_SIDE_OPPONENT].toxicSpikesAmount == amountUsed);
    }
}
```